### PR TITLE
Completed forgot password feature

### PR DESCRIPTION
**Pull request by harshadajog**
### :scroll: Description

- Describe the general shape of this PR (new feature? refactor? bug fix? one-line change?)
New Feature

- Describe what changes are being made
1. Added ForgotPassword & SetNewPassword components.
2. Added GraphQL query & mutations Get_Token, update_token and reset_password.
3. Email sent on forgot password request.

- Describe why these changes are being made
https://trello.com/c/XC1mShgs/101-eng-forgot-password

- List the use cases and edge cases relevant to this PR
1. User clicks forgot password link.
2. User taken to reset password screen.
3. Cases - User exists in system. Negative - User not registered in system
4. Valid user requests send password reset email.
5. User clicks email link in inbox
6. Email link cases:
    valid,
    null,
    invalid,
    expired
7. If valid, user taken to set new password screen.
8. if Invalid, error shown (set new password screen is not shown)
9. if password reset success, success message shown
10. If error, error is displayed instead.

- Describe how errors will be handled. How will we know if this code breaks in production
Backend - try...catch. Logger.error/debug messages added
Frontend: handle all conditions

- Describe any external libraries/dependencies added or removed in this PR
None

- Describe any security risks are there regarding this change
The password token URL expires after 20 minutes.
But if anyone gets access to it, he can change password

- Describe how you tested these changes
Manual, Functional

- Link to relevant external documentation
None

--------------------------
### :clipboard: Mandatory Checklist

- [x] Example of a checked item (please remove when creating your Pull Request) 

- [x] Linked to the Github Issues being addressed using the right sidebar :arrow_right:
- [x] Have you discussed these changes with the project leader(s)?
- [x] Do all variable and function names communicate what they do?
- [x] Were all the changes commented and / or documented?
- [x] Is the PR the right size? (If the PR is too large to review, it might be good to break it up into multiple PRs.)
- [x] Does all work in progress, temporary, or debugger code have a TODO comment with links to Github issues?
- [x] *If you changed the user interface, did you add before and after screenshots to below?*

--------------------------
#### After

image

image

https://user-images.githubusercontent.com/3578183/141852832-d6478c6b-e36b-474b-baa4-b5ab8a811af8.mp4

--------------------------
### :blue_book: Glossary
- PR = Pull Request

### CODE REVIEW

File: frontend/src/components/ForgotPassword/ForgotPasswordRequest.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react';
+import {NavLink} from 'react-router-dom';
+import { useMutation, useLazyQuery } from '@apollo/client';
+import { useTranslation } from 'react-i18next';
+import { GET_ACCOUNT_BY_EMAIL, FORGOT_PASSWORD } from '../../graphql/graphql';
+import {
+    isNullOrEmpty, validEmail
+  } from '../../utils/validations';
+
+  import './ForgotPasswordRequest.scss';
```
Comment (jdunbrack) [inline frontend/src/components/ForgotPassword/ForgotPasswordRequest.jsx:10]: Stray indent

File: frontend/src/components/ForgotPassword/SetNewPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import { useMutation, useLazyQuery } from '@apollo/client';
+import parseJwt from '../../utils/parseJwt';
+import {
+    isNullOrEmpty, isPasswordValid,
+  } from '../../utils/validations';
+  import { RESET_PASSWORD, GET_RESET_PASSWORD_TOKEN } from '../../graphql/graphql';
+
+  import './ForgotPasswordRequest.scss';
+
+function SetNewPassword (){
```
Comment (jdunbrack) [inline frontend/src/components/ForgotPassword/SetNewPassword.jsx:15]: Indentation is inconsistent inside function definition

File: frontend/src/components/ForgotPassword/SetNewPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+import { useTranslation } from 'react-i18next';
+import queryString from 'query-string';
+import { useMutation, useLazyQuery } from '@apollo/client';
+import parseJwt from '../../utils/parseJwt';
+import {
+    isNullOrEmpty, isPasswordValid,
+  } from '../../utils/validations';
+  import { RESET_PASSWORD, GET_RESET_PASSWORD_TOKEN } from '../../graphql/graphql';
+
+  import './ForgotPasswordRequest.scss';
```
Comment (jdunbrack) [inline frontend/src/components/ForgotPassword/SetNewPassword.jsx:13]: Stray Indents

**Review by jdunbrack — CHANGES_REQUESTED** [review]
A few quick formatting/indentation changes, but everything else looks good.

File: frontend/src/components/ForgotPassword/SetNewPassword.jsx (hunk 2 above)
Comment (harshadajog) [reply frontend/src/components/ForgotPassword/SetNewPassword.jsx:15]: Fixed

File: frontend/src/components/ForgotPassword/SetNewPassword.jsx (hunk 3 above)
Comment (harshadajog) [reply frontend/src/components/ForgotPassword/SetNewPassword.jsx]: Fixed in latest commit

File: frontend/src/components/ForgotPassword/ForgotPasswordRequest.jsx (hunk 1 above)
Comment (harshadajog) [reply frontend/src/components/ForgotPassword/ForgotPasswordRequest.jsx]: Fixed

**Comment by harshadajog** [comment]
Fixed the indents. Will look into why ESLint didn't report the issues.

**Review by jdunbrack — APPROVED** [review]

**Review by JMStudiosJoe — APPROVED** [review]

**Pull request merged**